client/components/repository/common/DueDate.vue

```diff
@@ -0,0 +1,63 @@
+<template>
+  <div :class="warning" class="due-date d-flex align-center">
+    <v-icon v-if="icon" size="16" class="icon mr-1">mdi-clock-outline</v-icon>
+    <span class="text-no-wrap">{{ value | formatDate(format) }}</span>
+  </div>
+</template>
+
+<script>
+import { isAfterOrEqual, truncateTime } from '@/utils/date';
+import isAfter from 'date-fns/isAfter';
+import { mapGetters } from 'vuex';
+import sub from 'date-fns/sub';
+
+export default {
+  name: 'workflow-due-date',
+  inheritAttrs: false,
+  props: {
+    value: { type: String, required: true },
+    format: { type: String, default: null },
+    icon: { type: Boolean, default: false }
+  },
+  data: () => ({ now: new Date() }),
+  computed: {
+    ...mapGetters('repository', ['workflow']),
+    currentDate: vm => truncateTime(new Date()),
+    dueDate: vm => truncateTime(new Date(vm.value)),
+    didWarningThresholdElapse() {
```

Comment:
Since it is bound to workflow ignore the previous comment 🤡  and consider renaming the component to indicate that